contracts/orderbook/Logic/OrderBook.sol

```diff
@@ -0,0 +1,413 @@
+                    );
+            } else {
+                operand =
+                    order.amountReceived >=
+                    getDexRate(
+                        swapImpl,
+                        order.base,
+                        order.loanTokenAddress,
+                        payload,
+                        order.collateralTokenAmount
+                    );
+            }
+            if (operand) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function getDexRate(
+        ISwapsImpl swapImpl,
+        address base,
+        address loanTokenAddress,
+        bytes memory payload,
+        uint256 amountIn
+    ) public returns (uint256 rate) {
+        (rate, ) = swapImpl.dexAmountOutFormatted(
+            payload,
+            10**IERC20Metadata(base).decimals()
+        );
+        rate = (rate * amountIn) / 10**IERC20Metadata(base).decimals();
+    }
+
+    function priceCheck(
+        address loanTokenAddress,
+        address base,
+        ISwapsImpl swapImpl,
+        bytes memory payload
+    ) public returns (bool) {
+        uint256 dexRate = getDexRate(
+            swapImpl,
+            base,
+            loanTokenAddress,
+            payload,
+            10**IERC20Metadata(base).decimals()
+        );
+        uint256 indexRate = queryRateReturn(
+            base,
+            loanTokenAddress,
+            10**IERC20Metadata(base).decimals()
+        );
+        if (dexRate >= indexRate) {
+            if (((dexRate - indexRate) * 1000) / dexRate <= 5) {
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            if (((indexRate - dexRate) * 1000) / indexRate <= 5) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
+
+    function executeOrder(bytes32 orderID) external pausable {
+        require(!_allOrders[orderID].isCancelled, "OrderBook: non active");
+        IOrderBook.Order memory order = _allOrders[orderID];
+        address srcToken;
+        uint256 amountUsed;
+        (uint256 dexID, bytes memory payload) = _prepDexAndPayload(order.loanDataBytes);
+        if (payload.length == 0) {
+            if (order.orderType == IOrderBook.OrderType.LIMIT_OPEN) {
+                payload = abi.encode(order.loanTokenAddress, order.base);
+            } else {
+                payload = abi.encode(order.base, order.loanTokenAddress);
+            }
+            dexID = 1;
+        }
+        ISwapsImpl swapImpl = ISwapsImpl(
+            IDexRecords(getSwapAddress()).retrieveDexAddress(dexID)
+        );
+        srcToken = order.collateralTokenAmount > order.loanTokenAmount ? order.base : order.loanTokenAddress;
+        require(order.timeTillExpiration > block.timestamp, "OrderBook: Order Expired");
+        if (_allOrders[orderID].orderType == IOrderBook.OrderType.LIMIT_OPEN) {
```

Comment:
since  you have `order` loaded in memory. use it

Reply:
will change